ios/Classes/SwiftOptimoveFlutterSdkPlugin.swift

```diff
@@ -0,0 +1,366 @@
+import Flutter
+import UIKit
+import OptimoveSDK
+
+public class SwiftOptimoveFlutterSdkPlugin: NSObject, FlutterPlugin, FlutterStreamHandler {
+    
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let channel = FlutterMethodChannel(name: "optimove_flutter_sdk", binaryMessenger: registrar.messenger())
+        let optimoveFlutterPlugin = SwiftOptimoveFlutterSdkPlugin()
+        optimoveFlutterPlugin.initOptimove(registrar: registrar)
+        registrar.addMethodCallDelegate(optimoveFlutterPlugin, channel: channel)
+    }
+
+    fileprivate var eventSink: FlutterEventSink?
+    
+    fileprivate func initOptimove(registrar: FlutterPluginRegistrar) {
+        //crash the app if the keys are not found
+        let topPath = Bundle.main.path(forResource: registrar.lookupKey(forAsset: "optimove.json"), ofType: nil)!
+        let jsonData = try! String(contentsOfFile: topPath).data(using: .utf8)
+        let optimoveKeys = try! JSONDecoder().decode(OptimoveKeys.self, from: jsonData!)
+        
+        let flutterEventChannel: FlutterEventChannel = FlutterEventChannel(name: "optimove_flutter_sdk_events", binaryMessenger: registrar.messenger())
+        flutterEventChannel.setStreamHandler(self)
+        self.initOptimove(from: optimoveKeys)
+    }
+    
+    fileprivate func initOptimove(from optimoveKeys: OptimoveKeys){
+        let config = OptimoveConfigBuilder(optimoveCredentials: optimoveKeys.optimoveCredentials, optimobileCredentials: optimoveKeys.optimobileCredentials)
+        
+        if (optimoveKeys.enableDeferredDeepLinking) {
+            let dlHandler: DeepLinkHandler = { deepLinkResolution in
+                self.emitDeeplinkResolved(deepLinkResolution: deepLinkResolution)
+            }
+            if let cname = optimoveKeys.cname {
+                config.enableDeepLinking(cname: cname, dlHandler)
+            } else {
+                config.enableDeepLinking(dlHandler)
+            }
+        }
+        
+        if #available(iOS 10, *) {
+            config.setPushReceivedInForegroundHandler(pushReceivedInForegroundHandlerBlock: { notification , UNNotificationPresentationOptions -> Void in
+                self.emitPushNotificationReceivedEvent(pushNotification: notification)
+            })
+        }
+        
+        config.setPushOpenedHandler(pushOpenedHandlerBlock: { notification in
+            self.emitPushNotificationOpenedEvent(pushNotification: notification)
+        })
+        
+        if optimoveKeys.inAppConsentStrategy != .disabled {
+            config.enableInAppMessaging(inAppConsentStrategy: optimoveKeys.inAppConsentStrategy == .autoEnroll ? .autoEnroll : .explicitByUser)
```

Comment:
Should we explicitly map this in a switch (in case any new strategies are added in future)?

Reply:
Although going with a switch will help us in case we add more strategies,  I suggest we won't change it at this stage, especially when there is just one disabled in the near future